kclvm/tools/src/LSP/src/request.rs

```diff
@@ -120,7 +136,10 @@ pub(crate) fn handle_semantic_tokens_full(
 ) -> anyhow::Result<Option<SemanticTokensResult>> {
     let file = file_path_from_url(&params.text_document.uri)?;
     let path = from_lsp::abs_path(&params.text_document.uri)?;
-    let db = snapshot.get_db(&path.clone().into())?;
+    let db = match snapshot.try_get_db(&path.clone().into())? {
```

Comment:
Is this the expected optimistic lock? What are the number and interval of retries?

Reply:
this is pessimistic lock. Optimistic locking is not applicable here yet